src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectionPool/ConnectionSetupDiagnostics.cs

```diff
@@ -0,0 +1,86 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+
+namespace System.Net.Http
+{
+    internal static class ConnectionSetupDiagnostics
```

Comment:
In other libs we start the activity together with recording an `EventSource` or emitting a metric.
Since there is no such strong correlation between HTTP connection telemetry pillars & unlike `HttpTelemetry` this class doesn't have to be built against WASM, separation makes more sense to me.

Reply:
PS: note that metrics are also handled in separate classes. In HTTP there is a lot of telemetry logic in total, IMO it would be bad to unify it in a single monster class.

Reply:
Agree with the renaming, disagree with code organization preferences 😆